src/Cogs/Operations.py

```diff
@@ -426,6 +439,29 @@ async def add_signup(op: dict, sign_up_name, main_role, alt_role: str = None) ->
         op["Signed"] += 1
         return op
 
+    async def add_any_signup(self, op: dict, sign_up_name) -> dict:
+        """
+        Adds a user with given name to any role as they are available (Dps > Healer > Tank)
+        Note: No changes are made if all roles are full.
+        :param op: The operation to be updated.
+        :param sign_up_name: The user's nick name.
+        :return: dict: The updated operation. 
+        """
+        roles = ["Dps", "Healer", "Tank"]
+        name = f"{sign_up_name} (Any)"
+
+        for role in roles:
+            if (not await self.check_role_full(op, role)):
+                roles.remove(role)
+                op["Sign-ups"][role] += [name]
+
+                for r in roles:
+                    op["Sign-ups"][f"Alternate_{r}"] += [sign_up_name]
+                return op
+
+        # if it gets here -> all roles were full & nothing changed
```

Comment:
This should never happen, as we check for the op being full -> sign up as reserve. But could be an out of sync problem for op["Signed"] if that check breaks or this function is used elsewhere